From 24e0d618794a52f9c409eb339ad90970f7fcb65f Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Thu, 18 Oct 2018 15:47:19 +0800
Subject: [PATCH 22/24] pm: ignore restorecon failure

On the 9p filesystem, restorecon won't work. It causes apk can't be
installed. Just ignore the errors to workaround it.
---
 .../java/com/android/server/pm/PackageInstallerService.java     | 2 +-
 .../core/java/com/android/server/pm/PackageManagerService.java  | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/pm/PackageInstallerService.java b/services/core/java/com/android/server/pm/PackageInstallerService.java
index 0b32d1a5d69b..b1caf7d3f0b4 100644
--- a/services/core/java/com/android/server/pm/PackageInstallerService.java
+++ b/services/core/java/com/android/server/pm/PackageInstallerService.java
@@ -654,7 +654,7 @@ public class PackageInstallerService extends IPackageInstaller.Stub {
         }
 
         if (!SELinux.restorecon(stageDir)) {
-            throw new IOException("Failed to restorecon session dir: " + stageDir);
+            Slog.e(TAG, "Failed to restorecon session dir: " + stageDir);
         }
     }
 
diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index 82ad46f08aed..09e77212885a 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -15989,7 +15989,6 @@ public class PackageManagerService extends IPackageManager.Stub
 
             if (!SELinux.restoreconRecursive(afterCodeFile)) {
                 Slog.w(TAG, "Failed to restorecon");
-                return false;
             }
 
             // Reflect the rename internally
-- 
2.17.1

